### PR TITLE
DDF-4180 Move subscription interface into catalog-ui-search-api bundle

### PR DESCRIPTION
**Pull request by Bdthomson**
#### What does this PR do?
- Moves SubscriptionPersistentStore interface into `catalog-ui-search-api` bundle and exports it.
- Publishes the `WorkspacePersistentStore` as a service.
- Removes the cron string config for the `WorkspaceQueryService`.
- Adds new detailed exception to instantiation of an SMTP client session instead of allowing a NPE to get thrown.
- Removes `platform-util-unavailableurls` from `catalog-ui-search` pom because it no longer has to be embedded.

#### Who is reviewing it? 

@rzwiefel 
@Schachte 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@millerw8

#### How should this be tested?

Full build and installation of `profile-standard` as well as making sure the UI still loads should be enough.

#### What are the relevant tickets?
[DDF-4180](https://codice.atlassian.net/browse/DDF-4180)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/subscription/SubscriptionConstants.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.subscription;
+
+public class SubscriptionConstants {
+
+  private SubscriptionConstants() {}
+
+  public static final String SUBSCRIPTIONS_TYPE = "subscriptions";
+}
```
Comment (hayleynorton) [inline catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/subscription/SubscriptionConstants.java]: ❓ What is the benefit of creating a whole new class for one string constant vs. leaving it in `SubscriptionsPersistentStore`?

**Comment by Bdthomson** [comment]
build now

File: catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/subscription/SubscriptionConstants.java (hunk 1 above)
Comment (rzwiefel) [reply catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/subscription/SubscriptionConstants.java]: @hayleynorton @Bdthomson 
There can be valid uses for both constants in the interface and in a separate file.  If the string is something that helps define the interface/contract then i would say it might belong in the interface. 

 If we look at this case closer, `SUBSCRIPTIONS_TYPE` is used  as the key of the type to store these objects under in the `PersistentStore`. There is also a `SUBSCRIPTIONS_TYPE` duplicated in `PersistentStore`. So some of the other persistent stores use the constants directly out of the persistent store. Since there's only one persistent store and technically consumers of it could step on each others toes if two services were to try and use the same key then we should probably just remove the key entirely from the `SubscriptionPersistentStore` and use the one from `PersistentStore`. People could still step on each others toes by using the same key, but at least using the ones in the `PersistentStore` means we have a single place to store all of these and people will have to see that there's one already in there before accidentally using it. 

So TL;DR- Maybe remove this constant and just use the `PersistentStore` constants inside the implementation of `SubscriptionPersistentStoreImpl`

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceServiceImpl.java [hunk 2]
```diff
@@ -223,7 +223,8 @@ private QueryRequest getQueryRequestFromIds(Set<String> ids) {
 
   private List<Map<String, Object>> findSubscriptions() {
     try {
-      return persistentStore.get(SubscriptionConstants.SUBSCRIPTIONS_TYPE, "", 0, maxSubscriptions);
+      return persistentStore.get(
+          PersistenceType.SUBSCRIPTION_TYPE.toString(), "", 0, maxSubscriptions);
```
Comment (rzwiefel) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceServiceImpl.java]: Ideally, we shouldn't be using raw persistent store anywhere since we have the subscriptionpersistentstore implementation that we are using to hide the PersistentStore... but since you're just fixing the strings its out of scope. Just something we can look for in the future.

**Comment by rzwiefel** [comment]
Changes made, build now

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
build now

**Comment by andrewzimmer** [comment]
Hero Successful
- Full build
- Installed profile-standard
- verified all UI bundles started and running
- verified UI loaded and was functional

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
build now (rebase)
